src/lua/api-gateway/validation/validator.lua

```diff
@@ -30,22 +30,23 @@
 --   1. api-gateway-redis upstream needs to be set
 --   2. api-gateway-redis-replica needs to be set
 --
-local base              = require "api-gateway.validation.base"
-local redis             = require "resty.redis"
-local RedisHealthCheck  = require "api-gateway.redis.redisHealthCheck"
-local cjson             = require "cjson"
-local debug_mode        = ngx.config.debug
+local base = require "api-gateway.validation.base"
+local cjson = require "cjson"
+local debug_mode = ngx.config.debug
+
+local RedisConnectionProvider = require "api-gateway.redis.redisConnectionProvider"
+
+-- redis endpoints are assumed to be global per GW node and therefore are read here
+local redis_RO_upstream = "api-gateway-redis-replica"
+local redis_RW_upstream = "api-gateway-redis"
 
 -- class to be used as a base class for all api-gateway validators --
 local BaseValidator = {}
-local redisHealthCheck = RedisHealthCheck:new({
-    shared_dict = "cachedkeys"
-})
+
+local redisConnectionProvider = RedisConnectionProvider:new()
 
 function BaseValidator:new(o)
     local o = o or {}
-    self.redis_RO_upstream = self.redis_RO_upstream or "api-gateway-redis-replica"
-    self.redis_RW_upstream = self.redis_RW_upstream or "api-gateway-redis"
     setmetatable(o, self)
```

Comment:
There were two lines above 
self.redis_RO_upstream = self.redis_RO_upstream or "api-gateway-redis-replica"		
 self.redis_RW_upstream = self.redis_RW_upstream or "api-gateway-redis"

should still be here in order to support sepparate redises for oauth and api_key